test/upsample.jl

```diff
@@ -1,3 +1,15 @@
+@testset "upsample_nearest, integer scale via reshape" begin
+    x = reshape(Float32[1. 2.; 3. 4.], (2,2,1,1))
+    @test upsample_nearest(x, (3,3))[1,:] == [1,1,1, 2,2,2]
+
+    y = upsample_nearest(x, (2,3))
+    @test size(y) == (4,6,1,1)
+    ∇upsample_nearest(y, (2,3)) == [6 12; 18 24]
+gradtest(x -> upsample_nearest(x, (2,3)), rand(2,2,1,1), check_rrule=true)
```

Comment:
```suggestion

    gradtest(x -> upsample_nearest(x, (2,3)), rand(2,2,1,1), check_rrule=false)

```

Reply:
ah right, cannot check the closure, doesn't matter anyway, zygote testing is enough